src/main/java/org/jitsi/jigasi/transcription/OracleTranscriptionService.java

```diff
@@ -0,0 +1,353 @@
+    {
+        return true;
+    }
+
+    public class OracleStreamingSession implements StreamingRecognitionSession, OracleRealtimeClientListener
+    {
+        private OracleRealtimeClient client;
+
+        private UUID uuid;
+
+        private boolean sessionEnding = false;
+
+        private final String websocketUrl;
+
+
+        /**
+         * List of TranscriptionListeners which will be notified when a
+         * result comes in
+         */
+        private final List<TranscriptionListener> listeners = new ArrayList<>();
+
+        private Instant sessionStart;
+
+
+        public OracleStreamingSession(String websocketUrl)
+        {
+            this.websocketUrl = websocketUrl;
+            uuid = Generators.timeBasedReorderedGenerator().generate();
+            try
+            {
+                client = new OracleRealtimeClient(
+                        this,
+                        authProvider,
+                        compartmentId);
+            }
+            catch (Exception e)
+            {
+                logger.error("Error while creating OCI client", e);
+            }
+        }
+
+        private void connect()
+        {
+            if (client.isConnected())
+            {
+                return;
+            }
+
+            final RealtimeParameters realtimeClientParameters = RealtimeParameters.builder()
+                    .isAckEnabled(false)
+                    .finalSilenceThresholdInMs(1000)
+                    .build();
+            try
+            {
+                client.open(websocketUrl, 443, realtimeClientParameters);
+                sessionStart = Instant.now();
+            }
+            catch (Exception e)
+            {
+                logger.error("Error while connecting to OCI service", e);
+            }
+        }
+
+        @Override
+        public void sendRequest(TranscriptionRequest request)
+        {
+            connect();
+            try
+            {
+                client.sendAudioData(request.getAudio());
+            }
+            catch (Exception e)
+            {
+                logger.error("Error while sending audio data to OCI service", e);
+            }
+        }
+
+        @Override
+        public void end()
+        {
+            logger.info("Ending OCI session and waiting for the last results");
+            try
+            {
+                // Wait for 5 seconds for the last results to come in
+                CountDownLatch latch = new CountDownLatch(1);
+                latch.await(5, TimeUnit.SECONDS);
+            }
+            catch (InterruptedException e)
+            {
+                logger.warn("Interrupted while waiting for the last results", e);
+            }
+
+            try
+            {
+                client.close();
+            }
+            catch (Exception e)
+            {
+                logger.error("Error while closing the OCI connection", e);
+            }
+            sessionEnding = true;
```

Comment:
This is set after the 5 s timeout, so we continue to send request in that period. Is this intended?

Reply:
refactored